batch/batch/driver/instance_collection/pool.py

```diff
@@ -431,7 +433,9 @@ async def user_runnable_jobs(user, remaining):
                         '''
 SELECT job_id, spec, cores_mcpu
 FROM jobs FORCE INDEX(jobs_batch_id_state_always_run_cancelled)
-WHERE batch_id = %s AND state = 'Ready' AND always_run = 0 AND inst_coll = %s AND cancelled = 0
+INNER JOIN batch_updates
```

Comment:
```
mysql> EXPLAIN SELECT job_id, spec, cores_mcpu
    ->  FROM jobs FORCE INDEX(jobs_batch_id_state_always_run_cancelled)
    ->  INNER JOIN batch_updates
    ->  ON jobs.batch_id = batch_updates.batch_id AND jobs.update_id = batch_updates.update_id
    ->  WHERE jobs.batch_id = 1000000 AND state = 'Ready' AND always_run = 0 AND inst_coll = "standard" AND cancelled = 0 AND batch_updates.committed
    ->  LIMIT 5;
+----+-------------+---------------+------------+--------+---------------------------------------------------------------------+------------------------------------------+---------+----------------------------+------+----------+-------------+
| id | select_type | table         | partitions | type   | possible_keys                                                       | key                                      | key_len | ref                        | rows | filtered | Extra       |
+----+-------------+---------------+------------+--------+---------------------------------------------------------------------+------------------------------------------+---------+----------------------------+------+----------+-------------+
|  1 | SIMPLE      | jobs          | NULL       | ref    | jobs_batch_id_state_always_run_cancelled                            | jobs_batch_id_state_always_run_cancelled | 132     | const,const,const,const    |    1 |    10.00 | Using where |
|  1 | SIMPLE      | batch_updates | NULL       | eq_ref | PRIMARY,batch_id,batch_updates_committed,batch_updates_start_job_id | PRIMARY                                  | 12      | const,batch.jobs.update_id |    1 |    90.00 | Using where |
+----+-------------+---------------+------------+--------+---------------------------------------------------------------------+------------------------------------------+---------+----------------------------+------+----------+-------------+
2 rows in set, 1 warning (0.01 sec)
```

Reply:
Let's revert this now that the `commit_batch_update` procedure guarantees jobs in a pending update won't be `Ready`.